### ISSUE TITLE
[Bug] Cluster is failing to bootup if the cluster is killed after deleting a model

### ISSUE THREAD

**Issue by naveentatikonda** [opensearch-project/k-NN#619]
Killing a running cluster after deleting a knn model is failing to reboot the cluster with the following error due to an issue with parsing XContent in [Model Graveyard](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/indices/ModelGraveyard.java#L146) 
```
org.opensearch.bootstrap.StartupException: org.opensearch.common.xcontent.XContentParseException: [-1:350] Failed to parse list:  expecting START_ARRAY but got START_OBJECT
        at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:184) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:171) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:104) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.cli.Command.mainWithoutErrorHandling(Command.java:138) ~[opensearch-cli-2.4.0.jar:2.4.0]
        at org.opensearch.cli.Command.main(Command.java:101) ~[opensearch-cli-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:137) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:103) ~[opensearch-2.4.0.jar:2.4.0]
Caused by: org.opensearch.common.xcontent.XContentParseException: [-1:350] Failed to parse list:  expecting START_ARRAY but got START_OBJECT
        at org.opensearch.common.xcontent.support.AbstractXContentParser.skipToListStart(AbstractXContentParser.java:378) ~[opensearch-x-content-2.4.0.jar:2.4.0]
        at org.opensearch.common.xcontent.support.AbstractXContentParser.list(AbstractXContentParser.java:312) ~[opensearch-x-content-2.4.0.jar:2.4.0]
        at org.opensearch.knn.indices.ModelGraveyard.fromXContent(ModelGraveyard.java:146) ~[?:?]
        at org.opensearch.common.xcontent.NamedXContentRegistry$Entry.lambda$new$0(NamedXContentRegistry.java:76) ~[opensearch-x-content-2.4.0.jar:2.4.0]
        at org.opensearch.common.xcontent.NamedXContentRegistry.parseNamedObject(NamedXContentRegistry.java:166) ~[opensearch-x-content-2.4.0.jar:2.4.0]
        at org.opensearch.common.xcontent.support.AbstractXContentParser.namedObject(AbstractXContentParser.java:435) ~[opensearch-x-content-2.4.0.jar:2.4.0]
        at org.opensearch.cluster.metadata.Metadata$Builder.fromXContent(Metadata.java:1763) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.gateway.PersistedClusterStateService.lambda$loadOnDiskState$1(PersistedClusterStateService.java:450) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.gateway.PersistedClusterStateService.consumeFromType(PersistedClusterStateService.java:514) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.gateway.PersistedClusterStateService.loadOnDiskState(PersistedClusterStateService.java:449) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.gateway.PersistedClusterStateService.loadBestOnDiskState(PersistedClusterStateService.java:374) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.gateway.GatewayMetaState.start(GatewayMetaState.java:132) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.node.Node.start(Node.java:1164) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.Bootstrap.start(Bootstrap.java:339) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.Bootstrap.init(Bootstrap.java:413) ~[opensearch-2.4.0.jar:2.4.0]
        at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:180) ~[opensearch-2.4.0.jar:2.4.0]
        ... 6 more
```

**Comment by navneet1v** [comment]
@naveentatikonda can we close this issue, as I can see that we have merged the fix?

**Comment by naveentatikonda** [comment]
Closing this issue as the fix is merged to 2.4 and backported to main, 2.x, 2.2 and 2.3 branches
